### ISSUE TITLE
Keep window on last closed position

### ISSUE THREAD

**Issue by Da3n0n** [f3d-app/f3d#856]
**Context**
When F3D opens, the window is position is not set at all and left to the window manager to position it, unless `--position` is set.
It would be nice to be able automatically save the position the window was the last time and restore it

Same for size, aka `resolution`.

**Comments**
This require to actually save the info somewhere on disk. We do already save a cache so this could be added to a generic cache where we put info in a .json format, but more discussions needed.

**Notes**

Original issue below

these are obviously not necessity, but these two small things would make the experience so much better.

1.Dark Title bar Implementation. 

 https://learn.microsoft.com/en-us/windows/apps/desktop/modernize/apply-windows-themes#enable-a-dark-mode-title-bar-for-win32-applications

2. Open F3D Window in Last closed position or Open in Center of Current Monitor.

**Comment by Meakk** [comment]
1 should be fairly easy to implement, but it has to be done in VTK.  
2 would be interesting and that's not the first time someone ask for it.

**Comment by mwestphal** [comment]
Shouldn't 2 be handled by the window manager already ? It seems implementable but i'd rather not if window manager provide the feature.

**Comment by Meakk** [comment]
Definitely! We're always open to new ideas :)  
Please share @Da3n0n

**Comment by Da3n0n** [comment]
well i don't want to just keep pumping you with features and make it overwhelming, but here's a few more off the top of the head Add View options to the viewport for orthographic views. 

drag and drop hdri into viewport to load it in the viewport would be really amazing or load HDRI option in the viewport would be nice either.

i will jsut leave these for now but i cna keep going 

I love this project and  even if yous dony implement these features will be still using f3d  and supporting it .if I had more experience I would go ahead and try implement these features myself.

**Comment by mwestphal** [comment]
>    drag and drop hdri into viewport to load it in the viewport

Done in 2.1.0 release

**Comment by mwestphal** [comment]
Rework this issue into the keep window on last closed position

**Comment by Da3n0n** [comment]
Amazing Stuff Guys Really well down such a Fast Implementation of the HDRI Drag and Drop feature  Honestly Amazing i Love this Project So much.

**Comment by mwestphal** [comment]
Size should be saved as well.

**Comment by brendevr** [comment]
This would be especially useful because I am having issues on CachyOS with F3D opening on my secondary monitor in the bottom left. This issue would fix it.

**Comment by mwestphal** [comment]
> This would be especially useful because I am having issues on CachyOS with F3D opening on my secondary monitor in the bottom left. This issue would fix it.

Indeed, please note you can hardcode a position in your config file if needed.

**Comment by Aun-Abbas** [comment]
Hi, I also want to solve this issue, can you show me where is the config file? So that I can change the position? Also is it possible to save the last position, so that next time window open on last saved position?

**Comment by mwestphal** [comment]
> can you show me where is the config file?

https://f3d.app/docs/user/CONFIGURATION_FILE#locations

> Also is it possible to save the last position, so that next time window open on last saved position?

No, this issue is still open as you can see.

BTW, such questions should ideally be asked on discord:
https://discord.f3d.app

**Comment by Aun-Abbas** [comment]
Ok, thank you for your response.

**Comment by dreadbalaur** [comment]
> > This would be especially useful because I am having issues on CachyOS with F3D opening on my secondary monitor in the bottom left. This issue would fix it.
> 
> Indeed, please note you can hardcode a position in your config file if needed.

Following up on potential workaround: I know how to use the config file, but I could not find instructions on how to hardcode a position in there. Any help?

**Comment by mwestphal** [comment]
> > > This would be especially useful because I am having issues on CachyOS with F3D opening on my secondary monitor in the bottom left. This issue would fix it.
> > 
> > 
> > Indeed, please note you can hardcode a position in your config file if needed.
> 
> Following up on potential workaround: I know how to use the config file, but I could not find instructions on how to hardcode a position in there. Any help?

Answered on discord

**Comment by noclone** [comment]
Hi @mwestphal, I'd be glad to work on this

**Comment by mwestphal** [comment]
https://github.com/f3d-app/f3d/issues/3358 will be needed for this

**Comment by mwestphal** [comment]
Damn, it took us a while but here it is! @Aun-Abbas @dreadbalaur @Da3n0n next nightly should contain this feature!

**Comment by dreadbalaur** [comment]
Great news! Looking forward to testing it out.

**Comment by mwestphal** [comment]
It is available now: https://github.com/f3d-app/f3d/releases/tag/nightly

**Comment by dreadbalaur** [comment]
@mwestphal tested today with the current nightly build (published 1 hour ago at the time of typing) and I cannot replicate the behaviour.

- Windows 11 Pro 26200.8875 using F3D-3.5.0-241-g39235183-Windows-x86_64 (both exe and zip). PATH set for all users
- f3d persistently opens outside the screen. Taskbar preview shows a black screen. I have deleted %AppData folders in case there are leftover .json files in previous workarounds. Only way to retrieve the window so far is by using Windows Snap. Despite my attempt to retrieve the windows and reposition it on the screen, this behaviour repeats after each restart.

<img width="210" height="151" alt="Image" src="https://github.com/user-attachments/assets/2f8d405d-6213-49bb-8826-2b909c0fcc6d" />

**Comment by Meakk** [comment]
Do you have multiple monitors?

**Comment by dreadbalaur** [comment]
Yes. Behaviour persists even if I change projection mode to isolate either monitor, or disconnect multiple monitors entirely.

**Comment by mwestphal** [comment]
Ok, multiple monitor test needed. @noclone do you think you could check ?

**Comment by noclone** [comment]
not for at least a week sorry

**Comment by mwestphal** [comment]
I checked, cache is broken on windows fully, windows just is not visible on second startup.

**Comment by mwestphal** [comment]
Fixed by #3457

**Comment by mwestphal** [comment]
@dreadbalaur fixed, you can check the last nightly

**Comment by dreadbalaur** [comment]
@mwestphal working great so far. Thank you all.
